src/integrator_interface.jl

```diff
@@ -613,15 +613,13 @@ function check_error(integrator::DEIntegrator)
     # This implementation is intended to be used for ODEIntegrator and
     # SDEIntegrator.
     if isnan(integrator.dt)
-        if verbose
-            @warn("NaN dt detected. Likely a NaN value in the state, parameters, or derivative value caused this outcome.")
-        end
+        @SciMLMessage("NaN dt detected. Likely a NaN value in the state, parameters, or derivative value caused this outcome.", verbose, :dt_NaN)
```

Comment:
If `verbose` is a bool, does this have a fallback to continue working like before?